spec/unit/transaction/resource_harness_spec.rb

```diff
@@ -1,7 +1,7 @@
 #! /usr/bin/env ruby
 require 'spec_helper'
 
-require 'puppet/transaction/resource_harness'
```

Comment:
Ok, I'll add the requirement, although I suspect that we have a vast number of places where we indeed do rely on transitive dependencies.